### PR TITLE
Yhj/add bcewithlogits

### PR DESCRIPTION
**Pull request by yeyeye333**
## Motivation and Context
add bcewithlogits

## Description
add bcewithlogits

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/camb/functions/bce_with_logits.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        posWeightFlag = false;
+    }
+
+    std::vector<DiopiTensor*> inTensors{&gradOutputTensor, &inputTensor, &targetTensor};
+    DIOPI_CALL(autoCastTensorType(ctx, inTensors, {diopi_dtype_float16, diopi_dtype_float32}));
+    DiopiTensor gradOutputTensorTmp = *inTensors[0];
+    DiopiTensor inputTensorTmp = *inTensors[1];
+    DiopiTensor targetTensorTmp = *inTensors[2];
+    DiopiTensor gradInputTensorTmp = gradInputTensor;
+    DIOPI_CALL(dataTypeCast(ctx, gradInputTensorTmp, inputTensorTmp.dtype()));
```
Comment (yewentao256) [inline impl/camb/functions/bce_with_logits.cpp:139]: 这里直接require一个指定dtype的tensor就行，不用datatype cast

File: impl/camb/functions/bce_with_logits.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    if (!posWeight) {
+        posWeightFlag = false;
+    }
+
+    std::vector<DiopiTensor*> inTensors{&inputTensor, &targetTensor};
+    DIOPI_CALL(autoCastTensorType(ctx, inTensors, {diopi_dtype_float16, diopi_dtype_float32}));
+    DiopiTensor inputTensorTmp = *inTensors[0];
+    DiopiTensor targetTensorTmp = *inTensors[1];
+    DiopiTensor outTensorTmp = outTensor;
+    DIOPI_CALL(dataTypeCast(ctx, outTensorTmp, inputTensorTmp.dtype()));
```
Comment (yewentao256) [inline impl/camb/functions/bce_with_logits.cpp:44]: 这里直接require一个指定dtype的tensor就行，不用datatype cast

File: impl/camb/functions/bce_with_logits.cpp (hunk 1 above)
Comment (yeyeye333) [reply impl/camb/functions/bce_with_logits.cpp]: > 这里直接require一个指定dtype的tensor就行，不用datatype cast

已修改

File: impl/camb/functions/bce_with_logits.cpp (hunk 2 above)
Comment (yeyeye333) [reply impl/camb/functions/bce_with_logits.cpp]: > 这里直接require一个指定dtype的tensor就行，不用datatype cast

已修改

**Review by yewentao256 — APPROVED** [review]

**Review by yangbofun — APPROVED** [review]

**Pull request merged**